addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -373,17 +373,29 @@ def _import_partner(self, company_id, name, phone, email, vat, country_code=Fals
         partner = self.env['res.partner'] \
             .with_company(company_id) \
             ._retrieve_partner(name=name, phone=phone, email=email, vat=vat, domain=domain)
+        country_code = postal_address.get('country_code')
+        country = self.env.ref(f'base.{country_code.lower()}', raise_if_not_found=False) if country_code else self.env['res.country']
+        state_code = postal_address.get('state_code')
+        state = self.env['res.country.state'].search(
+            [('country_id', '=', country.id), ('code', '=', state_code)]
+        ) if state_code and country else self.env['res.country.state']
         if not partner and name and vat:
             partner_vals = {'name': name, 'email': email, 'phone': phone}
             if peppol_eas and peppol_endpoint:
                 partner_vals.update({'peppol_eas': peppol_eas, 'peppol_endpoint': peppol_endpoint})
-            country = self.env.ref(f'base.{country_code.lower()}', raise_if_not_found=False) if country_code else False
-            if country:
-                partner_vals['country_id'] = country.id
             partner = self.env['res.partner'].create(partner_vals)
             if vat and self.env['res.partner']._run_vat_test(vat, country, partner.is_company):
```

Comment:
With aboo's suggesting, in some cases, you're changing the value of country from False to empty recordset, did you make sure that was ok ?

Reply:
yes, have tested it. it will behave the same as accessing the fields of empty record set returns (e.g country_id.id) returns False